cpu/qn908x/include/gpio_mux.h

```diff
@@ -56,6 +56,16 @@ extern "C" {
 #error "GPIO_T_ADDR(GPIO_PIN(1, x)) must be the GPIOB address"
 #endif
 
+/**
+ * @brief   Return whether the given pin is a CSHW pin.
+ */
+#define GPIO_T_IS_HWCS(pin) (((pin) & 0xff00u) == 0x8000)
```

Comment:
Yes, "hardware" (meaning, automatic) is only supported on certain pins, depending on the SPI bus used. With the current implementation this is not that important because you can always toggle the CS GPIO pin, run a SPI transaction and toggle it back (which is done when non-HWCS values of gpio_t are used). However, this will be more relevant when we can do DMA to SPI.

I'm using the same "gpio_t" type to store two types of values:
 * a regular GPIO pins (port + number), and
 * a "hardware CS", which are values with the 0x8000 mask.

As before, any other value of gpio_t is invalid. This patch makes the few 0x8000 .. 0x8003 values valid ones and representing the HW CS function / pin.

This macro in particular tell whether a "GPIO_T" is a HWCS.